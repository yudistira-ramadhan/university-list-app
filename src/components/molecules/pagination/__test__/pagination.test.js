import React from 'react'
import ReactDOM from 'react-dom'
import Pagination from '../pagination'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Pagination/>, div)
})

it('renders pagination correctly', () => {
  const { getByTestId } = render(<Pagination />)
  expect(getByTestId('pagination')).toBeTruthy()
})

it('matches snapshot', () => {
  const tree = renderer.create(<Pagination />).toJSON()
  expect(tree).toMatchSnapshot()
})