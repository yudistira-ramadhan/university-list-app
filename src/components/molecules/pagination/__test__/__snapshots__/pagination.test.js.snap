// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="d-flex my-3"
  data-testid="pagination"
>
  <ul
    className="pagination"
  >
    <li
      className="previous disabled"
    >
      <a
        aria-disabled="true"
        aria-label="Previous page"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        prev
      </a>
    </li>
    <li
      className="active"
    >
      <a
        aria-current="page"
        aria-label="Page 1 is your current page"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        1
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        aria-label="Page 2"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        2
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        aria-label="Page 3"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        3
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        aria-label="Page 4"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        4
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        aria-label="Page 5"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        5
      </a>
    </li>
    <li
      className="break-me"
    >
      <a
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        ...
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        aria-label="Page 9"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        9
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        aria-label="Page 10"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        10
      </a>
    </li>
    <li
      className="next"
    >
      <a
        aria-disabled="false"
        aria-label="Next page"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        next
      </a>
    </li>
  </ul>
</div>
`;
